cmd/lxc-openvdc/main.go

```diff
@@ -0,0 +1,218 @@
+	_imgPath := flag.String("img-path", "127.0.0.1/images", "Image path")
+	_cachePath := flag.String("cache-path", "/var/cache/lxc", "Cache path")
+
+	flag.Parse()
+
+	dist := *_dist
+	release := *_release
+	arch := *_arch
+	rootfsPath = *_rootfs
+	containerPath = *_containerPath
+	containerName = *_containerName
+	imgPath = *_imgPath
+	cachePath := *_cachePath
+
+	lxcPath = "/usr/share/lxc/"
+	cacheFolderPath = filepath.Join(cachePath, dist, release, arch)
+	imgPath = filepath.Join(imgPath, dist, release, arch)
+
+	err := PrepareCache()
+	if err != nil {
+		log.WithError(err).Error("Failed preparing cache.")
+	}
+
+	err = SetupContainerDir()
+	if err != nil {
+		log.WithError(err).Error("Failed setting up container directory.")
+	}
+
+	err = GenerateConfig()
+	if err != nil {
+		log.WithError(err).Error("Failed generating lxc config file.")
+	}
+}
+
+func SetupContainerDir() error {
+	err := os.MkdirAll(rootfsPath, os.ModePerm)
+	if err != nil {
+		return errors.Wrapf(err, "Failed creating container folder.")
+	}
+
+	if rootfsPath != "" {
+		err = DecompressXz("rootfs.tar.xz", rootfsPath)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New("RootfsPath not set.")
+	}
+
+	return nil
+}
+
+func PrepareCache() error {
+
+	if Exists(cacheFolderPath) == false {
+		err := CreateCacheFolder(cacheFolderPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	if Exists(filepath.Join(cacheFolderPath, "meta.tar.xz")) == false {
+		err := GetFile("meta.tar.xz")
+		if err != nil {
+			return errors.Wrapf(err, "Failed downloading file.")
+		}
+		err = DecompressXz("meta.tar.xz", cacheFolderPath)
+		if err != nil {
+			return errors.Wrapf(err, "Failed decompressing file.")
+		}
+	}
+
+	if Exists(filepath.Join(cacheFolderPath, "rootfs.tar.xz")) == false {
+		err := GetFile("rootfs.tar.xz")
+		if err != nil {
+			return errors.Wrapf(err, "Failed downloading file.")
+		}
+	}
+
+	return nil
+}
+
+func GenerateConfig() error {
+	lxcCfgPath := filepath.Join(lxcPath, "config")
+	cfgPath := filepath.Join(containerPath, "config")
+	cacheCfgPath := filepath.Join(cacheFolderPath, "config")
+
+	f, err := ioutil.ReadFile(cacheCfgPath)
+	if err != nil {
+		return errors.Wrapf(err, "Failed reading file: %s.", cacheCfgPath)
+	}
+
+	s := "\n# Distribution configuration\n" + string(f[:])
+	s = strings.Replace(s, "LXC_TEMPLATE_CONFIG", lxcCfgPath, -1)
+
+	s += fmt.Sprintf(
+		`
+# Container specific configuration
+lxc.rootfs = %s
+lxc.utsname = %s
+ 
+# Network configuration
+lxc.network.type = veth
+lxc.network.flags = up
+lxc.network.link = virbr0
+`, rootfsPath, containerName)
+
+	b := []byte(s)
+
+	err = ioutil.WriteFile(cfgPath, b, 0644)
+	if err != nil {
+		return errors.Wrapf(err, "Failed writing to file: %s.", cfgPath)
+	}
+
+	return nil
+}
+
+func GetFile(fileName string) error {
+
+	filePath := filepath.Join(cacheFolderPath, fileName)
+
+	res, err := http.Get("http://" + imgPath + "/" + fileName)
```

Comment:
The reason I did it this way was because path.Join didn't work well with "http://". It treated it like a file path and removed one of the slashes.

Solved it this way though: https://play.golang.org/p/5F9yeOGUus